src/compiler/execution.jl

```diff
@@ -111,7 +111,7 @@ macro cuda(ex...)
                     $kernel_tt = Tuple{map(Core.Typeof, $kernel_args)...}
                     $kernel = $cufunction($kernel_f, $kernel_tt; $(compiler_kwargs...))
                     if $launch
-                        $kernel($(var_exprs...); $(call_kwargs...))
+                        $kernel($kernel_args...; $(call_kwargs...))
```

Comment:
Okay now I understand the issue, @maleadt previously we didn't need to differentiate through the cudaconvert to the kernel, but now we do and those haven't been added.

I think this (or something similar) ought resolve: https://github.com/JuliaGPU/CUDA.jl/pull/2476